WalletWasabi/Crypto/ZeroKnowledge/LinearRelation/Knowledge.cs

```diff
@@ -25,5 +19,13 @@ internal Knowledge(Statement statement, ScalarVector witness)
 
 		internal ScalarVector RespondToChallenge(Scalar challenge, ScalarVector secretNonces) =>
 			Equation.Respond(Witness, secretNonces, challenge);
+
+		internal void AssertSoundness()
```

Comment:
It may be useful to make it clear that it's only for tests (if it is):

```suggestion
        /// <summary>For testing purposes.</summary>
		internal void AssertSoundness()
```

Reply:
hmm, it can certainly be used outside of tests, but i suspect we likely won't... @lontivero what do you think?

Reply:
I think we will never use it outside of tests.